Signal/src/AppDelegate.m

```diff
@@ -766,12 +766,20 @@ - (void)application:(UIApplication *)application
 
 - (void)application:(UIApplication *)application didReceiveLocalNotification:(UILocalNotification *)notification {
     if (!self.isEnvironmentSetup) {
-        OWSFail(@"%@ ignoring %s because environment is not yet set up.", self.tag, __PRETTY_FUNCTION__);
+        OWSFail(
+            @"%@ ignoring %s because environment is not yet set up: %@.", self.tag, __PRETTY_FUNCTION__, notification);
+        return;
+    }
+    if (application.applicationState != UIApplicationStateActive) {
+        OWSFail(@"%@ ignoring %s because app is not yet active: %@.", self.tag, __PRETTY_FUNCTION__, notification);
         return;
     }
+    DDLogInfo(@"%@ %s %@", self.tag, __PRETTY_FUNCTION__, notification);
 
     [AppStoreRating preventPromptAtNextTest];
     [[PushManager sharedManager] application:application didReceiveLocalNotification:notification];
+    // We only want to receive a single local notification per launch.
+    [application cancelAllLocalNotifications];
```

Comment:
Do we need to reconcile this with NotificationManager/PushManager which keep some local state related to "current" local notifications?

Reply:
Done, although I think that stale state was harmless.